### ISSUE TITLE
ENH: tau with DynInstAPI

### ISSUE THREAD

**Issue by tylerjereddy** [LLNL/spack#30988]
As far as I can tell `spack` doesn't currently support building `tau` such that this feature is available: https://www.cs.uoregon.edu/research/tau/docs/newguide/bk03ch02s02.html

In short, it adds an extra dependency on `DyninstAPI`, and `spack` can install `dyninst`, but it is not listed as a dependency of `tau` at the moment, and I don't see an option that would enable this?

To make matters worse, I can't actually tell what version of `dyninst` is appropriate for a given version of `tau`, and it seems that (manually, anyway) recent versions of `dyninst` are not compatible, while several older versions look a bit more promising but are also deprecated and not easy to install with `spack` anymore.

cc @jrmadsen @gshipman @eugeneswalker @khuck perhaps

**Comment by khuck** [comment]
@tylerjereddy you're correct that the tau package doesn't have any support for configuring TAU with DynInstAPI support. We should fix that, if it's needed.

**Comment by tylerjereddy** [comment]
I can tell you the basics of my use case to determine if it would be useful:

- want to use runtime/`LD_PRELOAD` profiling approach because recompiling an entire Python machine learning stack with `tau`-specific compile options would likely be painful/impractical
- want to use this runtime-only profiling approach to trace MPI data transfer/activity between ranks/nodes and produce i.e., jumpshot-style diagrams of Tensorflow/Horovod running on multiple ranks/GPUs

If you can't really do that detailed MPI tracing with a default compile-options build of Tensorflow + Horovod, then I'm out of luck anyway unless I recompile the whole stack, so the use case may be less strongly justified.
